packages/go_router/test/test_helpers.dart

```diff
@@ -70,9 +70,24 @@ class GoRouterGoSpy extends GoRouter {
   Object? extra;
 
   @override
-  void go(String location, {Object? extra}) {
+  Future<T?> go<T extends Object?>(String location, {Object? extra}) {
     myLocation = location;
     this.extra = extra;
+    return Future<T?>.value();
+  }
+}
+
+class GoRouterGoAsyncSpy extends GoRouter {
```

Comment:
Why add a new class when you could fix `GoRouterSpy` to include the `extra` parameter? The only difference between these two classes is this line:

GoRouterGoSpy:

```
  @override
  Future<T?> go<T extends Object?>(String location, {Object? extra}) {
    myLocation = location;
    this.extra = extra;
    return Future<T?>.value();
  }
```

GoRouterGoAsyncSpy:

```
  @override
  Future<T?> go<T extends Object?>(String location, {Object? extra}) {
    myLocation = location;
    this.extra = extra;
    return Future<T?>.value(extra as T?);
  }
```